apps/site/test/site_web/views/schedule_view_test.exs

```diff
@@ -1055,6 +1070,76 @@ defmodule SiteWeb.ScheduleViewTest do
       refute tabs =~ "info-tab"
       refute tabs =~ "timetable-tab"
     end
+
+    test "includes shuttles tab, if flag set and current shuttle alert", %{conn: conn} do
+      ongoing_lifecycles = [:ongoing, :ongoing_upcoming]
+
+      ongoing_lifecycles
+      |> Enum.each(fn lifecycle ->
+        fake_repo_fn = fn _, _, _ ->
+          [%Alerts.Alert{@shuttle_alert | lifecycle: lifecycle}]
+        end
+
+        tabs =
+          conn
+          |> put_req_header("cookie", "shuttles=true")
+          |> fetch_cookies
+          |> assign(:route, %Route{id: "83", type: 3})
+          |> assign(:tab, "alerts")
+          |> assign(:tab_params, [])
+          |> assign(:date_time, DateTime.utc_now())
+          |> assign_alerts(filter_by_direction?: true, repo_fn: fake_repo_fn)
+          |> route_header_tabs()
+          |> safe_to_string()
```

Comment:
Can we extract this lengthy stack of prerequisites to a helper function(s) within the test file? In particular I think it would be good to have an explicit `enable_shuttles_laboratory` function so we know what that logic is doing and can cleanly remove it when we get rid of the flag.